rest/sync_fn_test.go

```diff
@@ -327,6 +327,60 @@ func TestSyncFunctionErrorLogging(t *testing.T) {
 	assert.Equal(t, numErrors+1, numErrorsAfter)
 }
 
+func TestSyncFunctionException(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelInfo, base.KeyHTTP, base.KeyJavascript)
+
+	rtConfig := RestTesterConfig{
+		SyncFn: `
+		function(doc) {
+			if (doc.throwException) {
+				channel(undefinedvariable);
+			}
+			if (doc.throwExplicit) {
+				throw("Explicit exception");
```

Comment:
Can you include a case for our documented use of throw to reject documents:
[ throw({forbidden: "read only!"})](https://docs.couchbase.com/sync-gateway/current/sync-function.html#use-10) 
The stat shouldn't be incremented in this case.